platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java

```diff
@@ -16,46 +16,155 @@
+     */
+    public static final long GENESIS_ROUND = 0;
 
     private static final class ClassVersion {
         public static final int ORIGINAL = 1;
-        public static final int ADDED_PREVIOUS_ADDRESS_BOOK = 2;
     }
 
-    private static final class ChildIndices {
-        public static final int PLATFORM_DATA = 0;
-        public static final int ADDRESS_BOOK = 1;
-        public static final int PREVIOUS_ADDRESS_BOOK = 2;
-    }
+    /**
+     * The address book for this round.
+     */
+    private AddressBook addressBook;
+
+    /**
+     * The previous address book. A temporary workaround until dynamic address books are supported.
+     */
+    private AddressBook previousAddressBook;
+
+    /**
+     * The round of this state. This state represents the handling of all transactions that have reached consensus in
+     * all previous rounds. All transactions from this round will eventually be applied to this state. The first state
+     * (genesis state) has a round of 0 because the first round is defined as round 1, and the genesis state is before
+     * any transactions are handled.
+     */
+    private long round = GENESIS_ROUND;
+
+    /**
+     * running hash of the hashes of all consensus events have there been throughout all of history, up through the
+     * round received that this SignedState represents.
+     */
+    private Hash hashEventsCons;
+
+    /**
+     * the consensus timestamp for this signed state
+     */
+    private Instant consensusTimestamp;
+
+    /**
+     * the minimum generation of famous witnesses per round
+     */
+    //    private List<MinGenInfo> minGenInfo; // TOD make sure it's ok to remove this
+
+    /**
+     * The version of the application software that was responsible for creating this state.
+     */
+    private SoftwareVersion creationSoftwareVersion;
+
+    /**
+     * The epoch hash of this state. Updated every time emergency recovery is performed.
+     */
+    private Hash epochHash;
+
+    /**
+     * The next epoch hash, used to update the epoch hash at the next round boundary. This field is not part of the hash
+     * and is not serialized.
+     */
+    private Hash nextEpochHash;
+
+    /**
+     * The number of non-ancient rounds.
+     */
+    private int roundsNonAncient;
+
+    /**
+     * A snapshot of the consensus state at the end of the round, used for restart/reconnect
+     */
+    private ConsensusSnapshot snapshot;
+
+    /**
+     * the time when the freeze starts
+     */
+    private Instant freezeTime;
+
+    /**
+     * the last freezeTime based on which the nodes were frozen
+     */
+    private Instant lastFrozenTime;
+
+    /**
+     * Data on node uptime.
+     */
+    private UptimeDataImpl uptimeData = new UptimeDataImpl();
 
     public PlatformState() {}
 
     /**
      * Copy constructor.
      *
-     * @param that
-     * 		the node to copy
+     * @param that the object to copy
      */
     private PlatformState(final PlatformState that) {
```

Comment:
Optional: add nullity annotations throughout this class

Reply:
Added. It's annoying, basically everything has to be `@Nullable` given the funky lifecycle of serialized objects.